src/Components/WorkloadsListTable/WorkloadsListTable.js

```diff
@@ -80,12 +85,49 @@ const WorkloadsListTable = ({
 
   const buildFilteredRows = (items) => {
     setRowsFiltered(false);
-    const filtered = items.filter((workloadData) => {
-      return passFilterWorkloads(workloadData, filters);
-    });
+    const filtered = items
+      .filter((workloadData) => {
+        return passFilterWorkloads(workloadData, filters);
+      })
+      .sort((a, b) => {
+        let fst, snd;
+        const d = filters.sortDirection === 'asc' ? 1 : -1;
+        switch (filters.sortIndex) {
+          case WORKLOADS_TABLE_CELL_NAME:
+            if (a?.cluster.display_name && b?.cluster.display_name) {
+              return (
+                d *
+                a?.cluster.display_name.localeCompare(b?.cluster.display_name)
+              );
+            } else {
+              return d * a?.cluster.uuid.localeCompare(b?.cluster.uuid);
+            }
+          case WORKLOADS_TABLE_CELL_RECOMMENDATIONS:
+            fst = a.metadata.recommendations || 0;
+            snd = b.metadata.recommendations || 0;
+            return fst > snd ? d : snd > fst ? -d : 0;
+          case WORKLOADS_TABLE_SEVERITY:
+            fst = Object.values(a.metadata.hits_by_severity || 0).reduce(
+              (a, b) => a + b,
+              0
+            );
+            snd = Object.values(b.metadata.hits_by_severity || 0).reduce(
+              (a, b) => a + b,
+              0
+            );
+            return fst > snd ? d : snd > fst ? -d : 0;
```

Comment:
You could do something like this

```suggestion
          case WORKLOADS_TABLE_SEVERITY:
            fst = a.metadata.hits_by_severity || { 1: 0, 2: 0, 3: 0, 4: 0 };
            snd = b.metadata.hits_by_severity || { 1: 0, 2: 0, 3: 0, 4: 0 };

            for (let i = 4; i > 0; i--) {
              if (fst[i] > snd[i]) return d;
              if (fst[i] < snd[i]) return -d;
            }

            return 0;
```

Reply:
Ok, gonna do that :)